<!-- This launch file launches a single mvBlueFox MLC usb 2.0 camera -->
<!-- You should provide the launch file with your camera serial number and
     calibration file -->

<launch>

  <!-- Serial number on the back of camera -->
  <arg name="serial" default="25000495"/>
  <!-- Calibration files are located under bluefox/config -->
  <arg name="calibration_file" default="calib_$(arg serial).yml"/>
  <!-- Options for mode are: master, slave and standalone -->
  <arg name="mode" default="standalone"/>
  <arg name="fps" default="20.0"/>
  <arg name="width" default="640"/>
  <arg name="height" default="480"/>
  <arg name="color" default="false"/>
  <arg name="binning" default="false"/>
  <arg name="expose_us" default="5000"/>
  <!-- White balance parameter, options are: none, indoor, outdoor -->
  <arg name="wbp" default="none"/>

  <node pkg="bluefox2" type="bluefox2_node" name="cam_$(arg serial)" output="screen">
    <param name="fps" type="double" value="$(arg fps)"/>
    <param name="gain" type="double" value="0.0"/>
    <param name="color" type="bool" value="$(arg color)"/>
    <param name="width" type="int" value="$(arg width)"/>
    <param name="height" type="int" value="$(arg height)"/>
    <param name="serial" type="string" value="$(arg serial)"/>
    <param name="frame_id" type="string" value="cam_$(arg serial)"/>
    <param name="binning" type="bool" value="$(arg binning)"/>
    <param name="auto_expose" type="bool" value="false"/>
    <param name="expose_us" type="int" value="$(arg expose_us)"/>
    <param name="white_balance" type="string" value="$(arg wbp)"/>
    <param name="mode" type="string" value="$(arg mode)"/>
    <param name="calibration_url" type="string" value="package://bluefox2/config/$(arg calibration_file)"/>
  </node>

</launch>
